### ISSUE TITLE
libtiff: failed to install libtiff. nix cannot download tiff_4.0.7-6.debian.tar.xz from any mirror.

### ISSUE THREAD

**Issue by saturday06** [NixOS/nixpkgs#26415]
## Issue description

failed to install libtiff. nix cannot download tiff_4.0.7-6.debian.tar.xz from any mirror.

### Steps to reproduce

```shell
$ LC_ALL=C date; nix-env --version; nix-channel --list; nix-channel --update; nix-env -i libtiff
Tue Jun  6 10:37:00 JST 2017
nix-env (Nix) 1.11.9
nixpkgs https://nixos.org/channels/nixpkgs-unstable
downloading Nix expressions from ‘https://d3g5gsiof5omrk.cloudfront.net/nixpkgs/nixpkgs-17.09pre108299.ec9a23332f/nixexprs.tar.xz’...
downloading ‘https://d3g5gsiof5omrk.cloudfront.net/nixpkgs/nixpkgs-17.09pre108299.ec9a23332f/nixexprs.tar.xz’... [8243/8656 KiB, 3856.5 KiB/s]
unpacking channels...
installing ‘libtiff-4.0.7’
these derivations will be built:
  /home/saturday06/.local/nix/store/faz2l3gzp1d5bwk7wp7yz3shxrqw12z3-tiff_4.0.7-6.debian.tar.xz.drv
  /home/saturday06/.local/nix/store/37c4lbb7cc8s28qjgzf8sivpz1ha11hw-libtiff-4.0.7.drv
building path(s) ‘/home/saturday06/.local/nix/store/6v6d1yg5iy5gin8crfbqm9sjmhhfq5j9-tiff_4.0.7-6.debian.tar.xz’

trying http://http.debian.net/debian/pool/main/t/tiff/tiff_4.0.7-6.debian.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   332  100   332    0     0    783      0 --:--:-- --:--:-- --:--:--   784
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404 Not Found
error: cannot download tiff_4.0.7-6.debian.tar.xz from any mirror
builder for ‘/home/saturday06/.local/nix/store/faz2l3gzp1d5bwk7wp7yz3shxrqw12z3-tiff_4.0.7-6.debian.tar.xz.drv’ failed with exit code 1
cannot build derivation ‘/home/saturday06/.local/nix/store/37c4lbb7cc8s28qjgzf8sivpz1ha11hw-libtiff-4.0.7.drv’: 1 dependencies couldn't be built
error: build of ‘/home/saturday06/.local/nix/store/37c4lbb7cc8s28qjgzf8sivpz1ha11hw-libtiff-4.0.7.drv’ failed
```

## Technical details

* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...)
```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 7.9 (wheezy)
Release:	7.9
Codename:	wheezy
```
* Nix version: (run `nix-env --version`)
```
$ nix-env --version
nix-env (Nix) 1.11.9
```
* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`)
```
$ nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion
"17.09pre108299.ec9a23332f"
```
* Sandboxing enabled: (run `grep build-use-sandbox /etc/nix/nix.conf`)
```
$ cd <nix installation prefix>
$ grep -r build-use-sandbox *
nix/store/bnhhal6k69yrkzk1yf6gi3nr44qdg22s-nixpkgs-17.09pre108299.ec9a23332f/nixpkgs/.github/ISSUE_TEMPLATE.md:* Sandboxing enabled: (run `grep build-use-sandbox /etc/nix/nix.conf`)
nix/store/bnhhal6k69yrkzk1yf6gi3nr44qdg22s-nixpkgs-17.09pre108299.ec9a23332f/nixpkgs/.github/PULL_REQUEST_TEMPLATE.md:    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
nix/store/bnhhal6k69yrkzk1yf6gi3nr44qdg22s-nixpkgs-17.09pre108299.ec9a23332f/nixpkgs/nixos/modules/services/misc/nix-daemon.nix:        build-use-sandbox = ${if (builtins.isBool cfg.useSandbox) then boolToString cfg.useSandbox else cfg.useSandbox}
share/doc/nix/manual/manual.html:    multi-user settings with untrusted users.</p></dd><dt><span class="term"><code class="literal">build-use-sandbox</code></span></dt><dd><p>If set to <code class="literal">true</code>, builds will be
```

**Comment by lheckemann** [comment]
Can confirm, URL is dead. This won't affect most users because it's available through binary caches, but it would probably be sensible not to rely on debian's package repositories for packages that may just change or disappear beneath our feet. This is a similar issue in nature to #25702 and it would be nice to avoid these…

**Comment by vcunat** [comment]
Addressed by #26452, but it will probably need picking to 17.03 as well.

**Comment by moxian** [comment]
http://http.debian.net/debian/pool/main/t/tiff/tiff_4.0.8-2.debian.tar.xz does not exist anymore either.

**Comment by moxian** [comment]
This seems to have been fixed by edf6176
